client/beefy/src/worker.rs

```diff
@@ -301,10 +303,10 @@ where
 		}
 	}
 
-	fn handle_vote(&mut self, round: (MmrRootHash, NumberFor<B>), vote: (Public, Signature)) {
+	fn handle_vote(&mut self, round: (Payload, NumberFor<B>), vote: (Public, Signature)) {
 		self.gossip_validator.note_round(round.1);
 
-		let vote_added = self.rounds.add_vote(round, vote);
+		let vote_added = self.rounds.add_vote(round.clone(), vote);
```

Comment:
I think it would be better to change `add_vote` to clone internally only when the vote is actually going to be added.